tilewe/tournament.py

```diff
@@ -90,67 +94,77 @@ def total_engines(self) -> int:
     
     @property
     def avg_scores(self) -> list[float]: 
         return [self.total_scores[i] / max(1, self.game_counts[i]) for i in range(self.total_engines)]
 
+    @property
+    def elo_delta(self) -> list[float]: 
+        return [self.elo_end[i] - self.elo_start[i] for i in range(self.total_engines)]
+
     @property
     def average_match_duration(self) -> float:
         return self.total_time / max(1, self.total_games)
     
     def get_matches_by_engine(self, engine: int) -> list[MatchData]:
-        filtered_matches = [x for x in self.match_data if engine in x.engines]
-        return filtered_matches
-    
-    def get_game_count_by_engine(self, engine: int) -> int:
-        return self.game_counts[engine]
-    
-    def get_wins_by_engine(self, engine: int) -> int:
-        return self.win_counts[engine]
-    
-    def get_win_rate_by_engine(self, engine: int) -> float:
-        return self.win_counts[engine] / max(1, self.game_counts[engine])
-    
-    def get_score_by_engine(self, engine: int) -> int:
-        return self.total_scores[engine]
-    
-    def get_avg_score_by_engine(self, engine: int) -> float:
-        return self.total_scores[engine] / max(1, self.game_counts[engine])
-    
-    def get_starting_elo_by_engine(self, engine: int) -> int:
-        return self.elo_start[engine]
-    
-    def get_ending_elo_by_engine(self, engine: int) -> int:
-        return self.elo_end[engine]
-    
-    def get_delta_elo_by_engine(self, engine: int) -> int:
-        return self.elo_end[engine] - self.elo_start[engine]
+        return [x for x in self.match_data if engine in x.engines]
     
     def get_engine_rankings_display(self, sort_by: str = 'elo_end', sort_dir: str = 'desc') -> str:
+        # verify the given sort property exists
         if not hasattr(self, sort_by):
             return f"Invalid sort field '{sort_by}', must specify a valid TournamentResults property"
+
+        # verify the given sort property is valid
         sort_attr: list = getattr(self, sort_by)
         if not isinstance(sort_attr, list) or len(sort_attr) <= 0:
             return f"Invalid sort field '{sort_by}', must specify a list of length >0"
         if not isinstance(sort_attr[0], int) and not isinstance(sort_attr[0], float):
             return f"Invalid sort field '{sort_by}', must specify a numeric list"
+
+        # verify the given sort direction
         if sort_dir != 'asc' and sort_dir != 'desc':
             return f"Invalid sort direction '{sort_dir}', try 'asc' or 'desc'"
 
+        # build the results table
+        N = self.total_engines
+        len_names = max(5, min(24, max([len(x) for x in self.engine_names]) + 1))
+        len_score = max(6, max([math.floor(math.log10(max(1, self.total_scores[i])) + 1) for i in range(N)]) + 1)
+        len_games = max(7, max([math.floor(math.log10(max(1, self.game_counts[i])) + 1) for i in range(N)]) + 1)
```

Comment:
The number at the front with `max(#` is effectively the minimum column width. If there is then a `min(#` that is effectively the maximum column width, otherwise no max width. Then there is a `max` of the data, usually with `+ 1` for one additional padding on the max data length